libdistopia/src/distopia.cpp

```diff
@@ -907,7 +1001,42 @@ namespace distopia {
         }
         return HWY_DYNAMIC_DISPATCH(CalcDistanceArrayTriclinicSingle)(a, b, na, nb, box, out);
     }
-
+    HWY_DLLEXPORT template <> void CalcSelfDistanceArrayNoBox(const float *a, int n, float *out) {
+        if (n < GetNFloatLanes()) {
+            return distopia::N_SCALAR::CalcSelfDistanceArrayNoBoxSingle(a, n, out);
+        }
```

Comment:
@richardjgowers do we still need to guard < full size with the strategy for overhangs?

Reply:
Yes. We're still aggressively loading full vectors